src/operator/linalg_impl.h

```diff
@@ -1234,6 +1234,137 @@ LINALG_GPU_SYEVD_WORKSPACE_QUERY(DnDsyevd, double)
 
 #endif  // __CUDACC__
 
+//////////////////////////////// GESVD ////////////////////////////////////////////
+
+// CPU/GPU-versions of LAPACK function "gesvd"
+
+template<typename xpu, typename DType> inline
+void check_gesvd(const Tensor<xpu, 2, DType>& UT,
+                 const Tensor<xpu, 1, DType>& L,
+                 const Tensor<xpu, 2, DType>& V) {
+  // Any checking that helps user debug potential problems.
+  CHECK_LE(V.size(0), V.size(1))
+    << "The second to last dimension of A must be less or equal to the "
+    << "last dimension";
+  CHECK_EQ(UT.size(0), UT.size(1))
+    << "UT must be square matrix";
+  CHECK_EQ(V.size(0), L.size(0))
+    << "V, L have incompatible sizes";
+  CHECK_EQ(V.size(0), UT.size(0))
+    << "V, UT must have compatible sizes";
+}
+
+#define LINALG_CPU_GESVD(fname, DType) \
+template<> inline \
+void linalg_gesvd<cpu, DType>(const Tensor<cpu, 2, DType>& UT, \
+                              const Tensor<cpu, 1, DType>& L, \
+                              const Tensor<cpu, 2, DType>& V, \
+                              const Tensor<cpu, 1, DType>& work, \
+                              Stream<cpu> *s) { \
+  check_gesvd(UT, L, V); \
+  DType lwork(0); \
+  MXNET_LAPACK_##fname(MXNET_LAPACK_ROW_MAJOR, V.size(0), V.size(1), \
+                       UT.dptr_, UT.stride_, L.dptr_, V.dptr_, V.stride_, \
+                       &lwork, -1); \
+  int ret(MXNET_LAPACK_##fname(MXNET_LAPACK_ROW_MAJOR, V.size(0), V.size(1), \
+                               UT.dptr_, UT.stride_, L.dptr_, V.dptr_, V.stride_, \
+                               work.dptr_, static_cast<int>(lwork))); \
+  CHECK_EQ(ret, 0) << #fname << " failed in lapack on cpu."; \
+}
+
+LINALG_CPU_GESVD(sgesvd, float)
+LINALG_CPU_GESVD(dgesvd, double)
+
+// Mangle temp storage requirements for DType and int into a single
```

Comment:
Remove this comment, it only applies to syevd. See my comment above, you have a single workspace here.

Reply:
Removed